docs/design-proposals/2020-09-07-canary-deployments-for-tidb-operator.md

```diff
@@ -0,0 +1,34 @@
+# Canary deployments for TiDB-Operator
+
+## Summary
+
+This document presents a design to support canary deployments TiDB-Operator.
+
+## Motivation
+
+When upgrade TiDB-Operator, users may want to deploy multiple cluster-scoped TiDB-Operators in the same Kubernetes cluster.
+And choose some specific tidb clusters will be managed by new  version TiDB-Operator, the others are still managed by old version. When the new version TiDB-Operator is fully tested, replace the old one.
+
+### Goals
+
+* Deploy multiple TiDB-Operator in one Kubernetes cluster
+* Manage tidbcluster which be selected
+* Don't manage `tidbcluster` which be excluded
+
+### Non-Goals
+
+* Upgrading `TiDB-Operator`
+* Upgrading `tidbcluster`
+
+## Proposal
+
+Add `labelSelector` and `excludedLabelSelector`
+When sync `tidbcluster` CRDs filter by `labelSelector` and  excluded by `excludedLabelSelector`.
+
+### Example
+1. Update the old version `TiDB-Operator` `spec.excludedLabelSelector` excluded `tidbcluster` which be labeled `tidb-operator/managed=v1.1.4`
```

Comment:
Is it OK we just use `labelSelector`? For example, with TiDB Operator v1.1.3, all of the clusters will be labeled as `tidb-operator/managed=v1.1.3`, and then deploy TiDB Operator v1.1.4, configure its `labelSelector` as `tidb-operator/managed=v1.1.4`.

Reply:
Yeah, I think just use `labelSelector` is OK.